File: Chapter11/com/packtpub/spark/extras/ConsumingData.scala
import org.apache.spark.sql.SparkSession

object ConsumingData {

  def main(args: Array[String]): Unit = {

    // Build a SparkSession in Local Mode
    val spark = SparkSession
      .builder()
      .master("local[2]")
      .appName("My Spark App")
      .getOrCreate()

    val animals = Seq("dog", "cat", "bear")

    val animalData = spark.sparkContext.parallelize(animals)

    animalData.foreach(animal => {
      println("I am now a distributed " + animal.concat("!"))
    })
  }
}

File: Chapter11/extras/ConsumingData.py
from pyspark.sql import SparkSession


# Function to read data in RDD
def read(animal): print("I am now a distributed " + animal + "!")


# Create a Spark Session
spark = SparkSession\
    .builder\
    .appName("My Spark App")\
    .master("local[2]")\
    .getOrCreate()

# create a sample set of data as an RDD
animals = spark.sparkContext.parallelize(["dog", "cat", "bear"])

# apply a function on the RDD
animals.foreach(read)
